quickwit-swim/src/membership.rs

```diff
@@ -119,7 +122,24 @@ impl ArtilleryMemberList {
         (suspect_members, down_members)
     }
 
-    pub fn mark_node_alive(&mut self, src_addr: &SocketAddr) -> Option<ArtilleryMember> {
+    // make sure node id is set correctly for the host (in case it changes)
+    pub fn ensure_node_id(&mut self, src_addr: SocketAddr, node_id: &str) {
+        for member in &mut self.members {
+            if member.remote_host() == Some(src_addr) {
+                member.node_id = node_id.to_string();
+            }
+        }
+    }
+    pub fn mark_node_alive(
+        &mut self,
+        src_addr: &SocketAddr,
+        node_id: String,
+    ) -> Option<ArtilleryMember> {
+        self.ensure_node_id(*src_addr, &node_id);
+        // remove self duplicates
+        let my_node_id = self.current_node_id();
+        self.members
+            .retain(|member| !(member.node_id() == my_node_id && member.remote_host().is_some()));
```

Comment:
I don't understand this. How does this remove self duplicates?

Reply:
Remove anything that has our id and a host set, because members with hosts are not self.

But I think this should be removed, we can avoid adding it in the first place.